File: demo/safira.rb
class Gemstone
  def initialize(props)
    @name = props[:name]
    @color = props[:color]
    @rarity = props[:rarity]
  end

  # Method to describe the gemstone
  def describe
    puts "The #{@name} gemstone is a beautiful shade of #{@color} and is #{@rarity}."
  end
end

# Create an object of the Gemstone class using props
my_gemstone = Gemstone.new({
  name: "safira",
  color: "blue",
  rarity: "rare"
})

# Describe the gemstone using object method
my_gemstone.describe

File: demo/safira.cs
using System;

class Gemstone
{
    public string Name { get; set; }
    public string Color { get; set; }
    public string Rarity { get; set; }

    public Gemstone(string name, string color, string rarity)
    {
        Name = name;
        Color = color;
        Rarity = rarity;
    }

    // Method to describe the gemstone
    public void Describe()
    {
        Console.WriteLine($"The {Name} gemstone is a beautiful shade of {Color} and is {Rarity}.");
    }
}

class Program
{
    static void Main(string[] args)
    {
        // Create a Gemstone object
        Gemstone myGemstone = new Gemstone("safira", "blue", "rare");

        // Describe the gemstone
        myGemstone.Describe();
    }
}
